test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -194,6 +254,106 @@ struct test
     }
 
 private:
+
+  template <typename Policy, typename T>
+  using TmpContainerType =
+#if TEST_DPCPP_BACKEND_PRESENT
+      std::conditional_t<__is_host_execution_policy<Policy>::value, std::vector<T>,
+                         std::vector<T, sycl::usm_allocator<T, sycl::usm::alloc::shared>>>;
+#else
+      std::vector<T>;
+#endif
+
+    // Test dangling iterators in return types for call with temporary data
+    template <int idx, typename Policy, typename Algo, typename ...Args>
+    constexpr void
+    test_dangling_pointers_arg_1(Policy&& exec, Algo&& algo, Args&& ...args)
+    {
+        // Check dangling iterators in return types for call with temporary data
+        if constexpr (!supress_dangling_iterators_check<std::remove_cvref_t<decltype(algo)>>)
+        {
+            using T = typename Container::__value_type;
+
+            // Check dangling with temporary containers in implementation
+            using res_ret_t = decltype(algo(CLONE_TEST_POLICY_IDX(exec, idx),
+                                            std::declval<TmpContainerType<decltype(exec), T>>(),
+                                            args...));
+
+            if constexpr (!std::is_fundamental_v<res_ret_t>)
+            {
+                if constexpr (!__all_dangling_in_result_v<res_ret_t>)
+                    res_ret_t::dummy;
```

Comment:
A more conventional way:

```suggestion
                    static_assert(!std::is_same_v<res_ret_t, res_ret_t>, "rest_ret_t is expected to be or consist of std::ranges::dangling");
```